docs/scaling-hive.md

```diff
@@ -0,0 +1,47 @@
+# Scaling Hive
+
+This document describes the main places users are likely to see scalability issues with Hive.
+
+## Hive uses CRDs!
+
+Most importantly, be aware that Hive uses CRDs to store its state. The amount of data that Hive can store is limited by the cluster's ability to store CRs. In OpenShift, this boils down to "how much data can we stuff into etcd before the cluster itself becomes unstable?" Best to not chance it. We recommend keeping the total number of Hive CRs in a Hive cluster in the thousands. Anything approaching 10,000 is the danger zone. In practice this means that a single Hive cluster can manage about 1000 clusters.
+
+# Horizontal vs. Vertical Scale
+
+With the exception of install pods (used only when clusters are installing), Hive 1.x is not horizontally scalable at the worker level. Most of the work Hive does happens in the hive-controllers pod, which is one single pod on one single worker. This means that when no installs are running, if you have a cluster with 10 workers, 9 of the workers are very bored. Hive clusters are prime candidates for using worker autoscaling. Keep the worker count as low as you can, but allow bursts of concurrent installs to call for temporary workers to spin up.
+
+In AWS, Hive performs best on c instance types. M instances are fine, but c instances are better.
+
+In general, the faster your etcd storage, the faster Hive will perform. c5.4xlarge has twice the EBS bandwidth as c5.2xlarge, and so Hive performs a bit better on c5.4xlarge (although in our opinion, not so much faster that c5.4xlarge is worth the premium over c5.2xlarge). In AWS, we recommend using io1 with 1200 iops for your masters.
+
+If you need to scale out the number of clusters you need to manage, use multiple Hive clusters -- approximately 1 Hive per 1000 clusters managed. Or fewer if you prefer to keep the blast radius smaller.
```

Comment:
I'd drop the blast radius comment, it's implied that of course you can manage less per hive, but it makes it sound like we explode a lot. :)

Reply:
It's a super-common microservices term
e.g. https://www.ibm.com/garage/method/practices/manage/practice_limited_blast_radius/

Reply:
I reworked it to emphasize architectures that are distributed into multiple regions or are cell-based. Hope this works.